be/src/exprs/vectorized/encryption_functions.cpp

```diff
@@ -199,4 +200,141 @@ ColumnPtr EncryptionFunctions::md5(FunctionContext* ctx, const Columns& columns)
     return result.build(ColumnHelper::is_all_const(columns));
 }
 
+Status EncryptionFunctions::sha2_prepare(FunctionContext* context, FunctionContext::FunctionStateScope scope) {
+    if (scope != FunctionContext::FRAGMENT_LOCAL) {
+        return Status::OK();
+    }
+
+    if (!context->is_constant_column(1)) {
```

Comment:
It should be supported when hash_length is not const.
We can do it row by row.
First get the hash_length value, and then create according Hash Digest to compute the first argument.